### PR TITLE
Add table/endpoints to enable/disable products/branches

### PR DESCRIPTION
**Pull request by mozbhearsum**
Very simple CRUD. Fully testable with the swagger UI.

### CODE REVIEW

File: api/src/shipit_api/api.yml [hunk 1]
```diff
@@ -229,6 +229,53 @@ paths:
...
+      parameters:
+      - name: product
+        in: query
+        required: true
+        type: string
+        enum:
+        - firefox
+        - fennec
+        - devedition
+        - thunderbird
```
Comment (rail) [inline api/src/shipit_api/api.yml]: Can we limit this to firefox and devedition for now? We are not going to use this for other products at this point.

File: api/src/shipit_api/api.yml [hunk 2]
```diff
@@ -229,6 +229,53 @@ paths:
...
+      - name: branch
+        in: query
+        required: true
+        type: string
```
Comment (rail) [inline api/src/shipit_api/api.yml]: Can you add limit this to beta and try only for now? Just in case.

File: api/src/shipit_api/api.yml [hunk 3]
```diff
@@ -416,3 +463,21 @@ definitions:
...
+    - branch
+    properties:
+      product:
+        type: string
+        example: firefox
+        enum:
+        - firefox
+        - fennec
+        - devedition
+        - thunderbird
```
Comment (rail) [inline api/src/shipit_api/api.yml]: The same here, firefix and devedition only.

File: api/src/shipit_api/api.yml [hunk 4]
```diff
@@ -416,3 +463,21 @@ definitions:
...
+      branch:
+        type: string
+        example: projects/maple
```
Comment (rail) [inline api/src/shipit_api/api.yml]: The same here, limit to beta and try

File: api/src/shipit_api/models.py [hunk 5]
```diff
@@ -229,3 +229,10 @@ def json(self):
             "completed": self.completed or "",
             "phases": [p.json for p in self.phases],
         }
+
+
+class DisabledProduct(db.Model):
+    __tablename__ = "shipit_api_disabled_products"
+
+    product = sa.Column(sa.String, nullable=False, primary_key=True)
```
Comment (rail) [inline api/src/shipit_api/models.py]: Not sure if using `primary_key` would let you add multiple branches for the same product and vice versa. Probably you want the combination to be unique.

File: api/src/shipit_api/api.py [hunk 6]
```diff
@@ -320,3 +321,33 @@ def phase_signoff(name, phase, uid):
     notify_via_irc(r.product, f"{phase} of {r.product} {r.version} build{r.build_number} signed off by {who}.")
 
     return dict(signoffs=signoffs)
+
+
+def get_disabled_products():
```
Comment (rail) [inline api/src/shipit_api/api.py]: We also need to auth-protect these methods.

File: api/src/shipit_api/models.py (hunk 5 above)
Comment (mozbhearsum) [reply api/src/shipit_api/models.py]: Setting `primary_key` on both columns turns it into a composite key. From https://docs.sqlalchemy.org/en/13/core/metadata.html:
> Multiple columns may be assigned the primary_key=True flag which denotes a multi-column primary key, known as a composite primary key.

File: api/src/shipit_api/api.py (hunk 6 above)
Comment (mozbhearsum) [reply api/src/shipit_api/api.py]: Just enable/disable, or also the getter?

**Comment by mozbhearsum** [comment]
The latest push should address everything, as long as we don't need to protect the list of disabled products/branches behind scopes.

File: api/src/shipit_api/api.py (hunk 6 above)
Comment (rail) [reply api/src/shipit_api/api.py]: get is readonly, no need for auth

**Comment by mozbhearsum** [comment]
@rail - this probably deserves another look after updating for openapi 3.0.

**Comment by rail** [comment]
Ooops, sorry, I should have waited before I merged my PR. :/

**Pull request merged**